### PR TITLE
component/BookCard

### LINKED ISSUE #33: BookCard
Build a BookCard component.

image

View the component in [Figma](https://www.figma.com/file/gGwSe7ilZCUmAIaJyUc4iY/Bristol-Tech-Fair-Design-System?node-id=535%3A3695)

### PR DESCRIPTION
**Pull request by shanesc**
Closes #33 

Adds the BookCard component from the Figma design.

### CODE REVIEW

**Comment by shanesc** [comment]
I'm not sure of the best way to pass images as props. If anyone has suggestions that would be amazing!

File: src/components/BookCard/BookCard.jsx [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+      <h3>{title}</h3>
+      <span>by {author}</span>
+    </BookCardWrapper>
+  );
+};
+
+BookCard.propTypes = {
+  title: PropTypes.string.isRequired,
+  author: PropTypes.string.isRequired,
+  cover: PropTypes.string.isRequired
```
Comment (merryface) [inline src/components/BookCard/BookCard.jsx]: using my limited knowledge, this seems fine, it's how I did the other TWO times I've used react...

**Review by merryface — DISMISSED** [review]
I can't really comment on the tests.
Didn't see anything obviously wrong (to my knowledge).
The only time I passed an image as a prop I did it this way too

**Comment by RobTF9** [comment]
The image prop needs to be a URL, as we'll most likely be serving images from an s3/cloudinary bucket.

Put the images you want to use for storybook in: `/public/assets/images/your-image.png` then pass the url to the component.

**Comment by shanesc** [comment]
> The image prop needs to be a URL, as we'll most likely be serving images from an s3/cloudinary bucket.
> 
> Put the images you want to use for storybook in: `/public/assets/images/your-image.png` then pass the url to the component.

Ok will do. Will we be pulling the URL from the database in production?

The prop passed in is being used as the source URL, is that correct? If so, I'll just move the image used for Storybook as above.

File: src/components/BookCard/BookCard.styles.js [hunk 2]
```diff
@@ -0,0 +1,41 @@
+import styled from 'styled-components';
+import { Link } from 'react-router-dom';
+
+export const BookCardWrapper = styled(Link)`
+  width: 24rem;
+  height: 30.9rem;
+  background: #ffffff;
```
Comment (RobTF9) [inline src/components/BookCard/BookCard.styles.js]: use var(--white) here

File: src/components/BookCard/BookCard.styles.js [hunk 3]
```diff
@@ -0,0 +1,41 @@
+import styled from 'styled-components';
+import { Link } from 'react-router-dom';
+
+export const BookCardWrapper = styled(Link)`
+  width: 24rem;
```
Comment (RobTF9) [inline src/components/BookCard/BookCard.styles.js]: Don't need a width.

File: src/components/BookCard/BookCard.styles.js [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+  height: 30.9rem;
```
Comment (RobTF9) [inline src/components/BookCard/BookCard.styles.js]: Don't need a height

**Review by RobTF9 — CHANGES_REQUESTED** [review]

**Comment by shanesc** [comment]
Made the requested changes

**Review by RobTF9 — APPROVED** [review]

**Pull request merged**